hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionInfo.java

```diff
@@ -346,15 +346,23 @@ static TableName getTable(final byte [] regionName) {
     return parseRegionName(regionName)[1];
   }
 
-  @InterfaceAudience.Private
-  static boolean isEncodedRegionName(byte[] regionName) throws IOException {
+  /**
+   * Figure if the passed bytes represent an encoded region name or not.
+   * @param regionName A Region name either encoded or not.
+   * @return True if <code>regionName</code> represents an encoded name.
+   */
+  @InterfaceAudience.Private // For use by internals only.
+  public static boolean isEncodedRegionName(byte[] regionName) throws IOException {
     try {
+      // TODO: This is extreme... a full parse and an exception if encoded -- and
+      // even then, we say it encoded just because it NOT encoded. Can we
+      // do something less expensive figuring if encoded?
       parseRegionName(regionName);
       return false;
     } catch (IOException e) {
-      if (StringUtils.stringifyException(e)
-      .contains(INVALID_REGION_NAME_FORMAT_MESSAGE)) {
-        return true;
+      if (StringUtils.stringifyException(e).contains(INVALID_REGION_NAME_FORMAT_MESSAGE)) {
```

Comment:
Can we make a version of parseRegionName that returns something instead of throwing? Then have both this call and the existing parseRegionName that needs to throw build on that?

Reply:
Let me.